components/core/src/os/process/linux.rs

```diff
@@ -12,17 +12,26 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use libc::{pid_t, c_int};
 use std::ffi::OsString;
 use std::path::PathBuf;
 use std::os::unix::process::CommandExt;
 use std::process::Command;
 
 use error::Result;
 
+extern "C" {
+    fn waitpid(pid: pid_t, status: *mut c_int, options: c_int) -> pid_t;
+}
+
 pub fn become_command(command: PathBuf, args: Vec<OsString>) -> Result<()> {
     become_exec_command(command, args)
 }
 
+pub fn wait_for_exit(pid: u32, status: *mut c_int) -> u32 {
+    return waitpid(pid, &mut status, 1 as c_int).unwrap as u32;
```

Comment:
If you leave off the `;`, you won't need the `return` as rust will return the result of the last expression.

Also, this should be wrapped in `unsafe {}` (rather than the later code in the supervisor.